tests/api_fastapi/views/public/test_connections.py

```diff
@@ -0,0 +1,63 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import pytest
+
+from airflow.models import Connection
+from airflow.utils.session import provide_session
+from tests.test_utils.db import clear_db_connections
+
+TEST_CONN_ID = "test_connection_id"
+TEST_CONN_TYPE = "test_type"
+
+
+@provide_session
+def _create_connection(session) -> None:
+    connection_model = Connection(conn_id=TEST_CONN_ID, conn_type=TEST_CONN_TYPE)
+    session.add(connection_model)
+
+
+class TestConnectionEndpoint:
+    @pytest.fixture(autouse=True)
+    def setup_attrs(self) -> None:
+        clear_db_connections(False)
+
+    def teardown_method(self) -> None:
+        clear_db_connections()
+
+    def create_connection(self):
+        _create_connection()
+
+
+class TestDeleteConnection(TestConnectionEndpoint):
+    @pytest.mark.db_test
```

Comment:
I think we can directly mark the entire file as `pytestmark = pytest.mark.db_test` because all of those tests will most likely need a db. (it's the same for datasets and dags).

We can fine tune later if we actually have tests that are not `db_test` in the `api_fastapi` folder.

Reply:
Nice one, make it class-wide. Indeed, most probably every test will definitely include some piece of db there